histoqc/BaseImage.py

```diff
@@ -90,38 +108,75 @@ def __setitem__(self, key, value):
 
         return super(BaseImage, self).__setitem__(key,value)
 
+    # setbounding box start coordinate and size
+    def setBBox(self):
+        # add self["img_bbox"] = (x, y, width, heigh)
+        osh = self["os_handle"]
+        # set default bbox
+        (dim_width, dim_height) = osh.dimensions
+        self["img_bbox"] = (0, 0, dim_width, dim_height)
+        # try to get bbox if bounding_box is ture
+        if self["bounding_box"]:
+            # try get bbox from os handle properties
+            x = osh.properties.get(openslide.PROPERTY_NAME_BOUNDS_X)
+            y = osh.properties.get(openslide.PROPERTY_NAME_BOUNDS_Y)
+            width = osh.properties.get(openslide.PROPERTY_NAME_BOUNDS_WIDTH)
+            height = osh.properties.get(openslide.PROPERTY_NAME_BOUNDS_HEIGHT)
+            if x and y and width and height:
+                try:
+                    self["img_bbox"] = tuple(int(num) for num in (x, y, width, height))
+                except:
+                    # no bbox info in slide set bounding_box as Flase
+                    self["bounding_box"] = False
+                    logging.warning(f"{self['filename']}: could not convert the bounding box")
+            else:
+                # no bbox info in slide set bounding_box as Flase
+                self["bounding_box"] = False
+                logging.warning(f"{self['filename']}: could not read the bounding box")
```

Comment:
nice, i like this.....maybe we *also* put a warning again the actual image so that we can see this in the tsv file?

something like
self["warnings"].append('Bounding Box requested but could not read')